File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.cpp
class Solution {
public:

    int count = 0;

    int path(TreeNode* root, int maxt) {

        if(root == nullptr)
            return 0;

        if(root->val >= maxt) {
            count = count + 1;
        }

        maxt = max(maxt, root->val);

        path(root->left, maxt);
        path(root->right, maxt);

        return count;
    }

    int goodNodes(TreeNode* root) {
        return path(root, root->val);
    }
};

File: Data Structures & Algorithms/permutation-string/submission-1.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        k = len(s1)

        l = [0] * 26
        l2 = [0] * 26

        left = 0

        for i in range(len(s1)):
            l[ord(s1[i]) - ord('a')] += 1

        for right in range(len(s2)):
            l2[ord(s2[right]) - ord('a')] += 1

            if right - left + 1 > k:
                l2[ord(s2[left]) - ord('a')] -= 1
                left += 1

            if right - left + 1 == k:
                if l2 == l:
                    return True

        return False